src/tests/End2EndTestsMain.cpp

```diff
@@ -14,7 +14,23 @@
 
 #include <gtest/gtest.h>
 
+extern bool gTestUsesWire;
+
 int main(int argc, char** argv) {
     testing::InitGoogleTest(&argc, argv);
+
+    for (int i = 1; i < argc; ++i) {
+        if (std::string("-w") == argv[i] || std::string("--use-wire") == argv[i]) {
```

Comment:
Just a style preference, but I'd use !strcmp() when comparing against const char *s, rather than constructing an std::string. But I see there's a ton of precedence here, so ignore me.

Reply:
Done, maybe we should have done that in the first place in the other spots too.